validator/sawtooth_validator/journal/publisher.py

```diff
@@ -74,9 +74,10 @@ def __init__(self,
         self._block_sender = block_sender
         self._pending_batches = []  # batches we are waiting for validation,
         # arranged in the order of batches received.
-        self._committed_txn = None  # Look-up cache for transactions that are
-        # committed in the current chain. Cache is used here so that we can
-        # support opportunistically building on top of a block we published.
+        self._committed_txn = TransactionCache(self._block_cache.block_store)
```

Comment:
This should be named as `_commited_txn_cache` or `_commited_txns`.  This name implies a single value.